tensorflow/python/distribute/all_reduce.py

```diff
@@ -25,7 +25,11 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import math_ops
-from tensorflow.python.ops import nccl_ops
+from tensorflow.python import pywrap_tensorflow
+if pywrap_tensorflow.IsBuiltWithROCm():
+  from tensorflow.contrib import rccl as nccl_ops
+else:
+  from tensorflow.python.ops import nccl_ops
```

Comment:
logic here doesn't seem to be related to this PR, and they are already present in `develop-upstream`. please do a rebase